log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+                }
+
+                if(!isIndexRemoved){
+                    // updated value NOT matched in allowed removable values
+                    allowedRemovableDataSet.splice((allowedRemovableDataSet.length - 1), 1);
+                }
+
+            } else {
+                //insert the new data
+                this.view.data(this.config.title).insert([data[i]]);
+                this.view.update();
+            }
+        }
+
+        var oldData;
+        var removeFunction = function(d) {
+            return d[xAxis] == oldData;
+        };
+
+        for (i = 0; i < allowedRemovableDataSet.length; i++) {
+            oldData = allowedRemovableDataSet[i];
+            this.view.data(this.config.title).remove(removeFunction);
+        }
+    } else{
+        for (i = 0; i < data.length; i++) {
+            var isValueMatched = false;
+            this.view.data(this.config.title).update(function(d) {
+                    if (color == null) {
+                        return  d[xAxis] == data[i][xAxis];
+                    } else {
+                        return  d[xAxis] == data[i][xAxis] &&  d[color] == data[i][color];
+                    }
+                },
+                yAxis,
+                function(d) {
+                    isValueMatched = true;
+                    return data[i][yAxis];
+                });
+
+            if(!isValueMatched){
+                this.view.data(this.config.title).insert([data[i]]);
+            }
+        }
+    }
+
+    //Group does not support duration update animation
+    if (this.config.mode == "group") {
+        this.view.update();
+    } else {
+        this.view.update({duration: 200});
+    }
+};
+
+bar.prototype.getSpec = function() {
+    return this.spec;
+};
+
+
+function getBarMark(config, metadata){
+    var markContent;
+    if (config.orientation == "left") {
+        markContent = {
+            "y": {"scale": "x", "field": metadata.names[config.x]},
+            "height": {"scale": "x", "band": true, "offset": -1},
+            "x": {"scale": "y", "field": metadata.names[config.y]},
+            "x2": {"scale": "y", "value": 0},
+            "fill": {"value": config.markColor},
+            "fillOpacity": {"value": 1}
+        };
+    } else {
+        markContent = {
+            "x": {"scale": "x", "field": metadata.names[config.x]},
+            "width": {"scale": "x", "band": true, "offset": -1},
+            "y": {"scale": "y", "field": metadata.names[config.y]},
+            "y2": {"scale": "y", "value": 0},
+            "fill": {"value": config.markColor},
+            "fillOpacity": {"value": 1}
+        };
+    }
+
+    var mark = {
+        "type": "rect",
+        "from": {"data": config.title},
+        "properties": {
+            "update": markContent,
+            "hover": {
+                "fillOpacity": {"value": 0.5}
+            }
+        }
+    };
+
+
+    return mark;
+}
+
+function getStackBarMark(config, metadata){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD